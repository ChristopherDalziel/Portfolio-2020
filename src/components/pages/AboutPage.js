import React from "react";
import { Link } from "react-router-dom";
import styled from "@emotion/styled";
import backgroundImage from "../../assets/Web_Final_Col.jpg";
import mobileBackgroundImage from "../../assets/Web_Final_Col.jpg";
import resume from "../../assets/res.pdf";
import Button from "../../utils/Button";

const AboutContainer = styled.main`
  height: 100vh;
  background-image: url(${backgroundImage});
  background-size: cover;
  @media (max-width: 815px) {
    background-image: url(${mobileBackgroundImage});
    overflow: hidden;
  }
`;

const AboutContent = styled.div`
  padding: 100px 0 0 50px;
  height: 80%;
  width: 30%;
  font-size: 15px;
  @media (max-width: 815px) {
    padding: 0;
    position: absolute;
    bottom: 0;
    overflow: hidden;
    height: 94vh;
    width: 100vw;
  }
`;

const AboutHeader = styled.h1`
  color: #493a37;
`;

class AboutPage extends React.Component {
  render() {
    return (
      <AboutContainer>
        <AboutContent>
          <AboutHeader>
            Hello. <br /> my names Chris.
          </AboutHeader>
          <br />
          <p>
            Over the past 6 years Iâ€™ve had the privilege of working as a store
            manager across 2 businesses gaining experience in many different
            collaborative environments and situations. 5 months ago, I stepped
            down from my position to chase a new passion with the hope of
            finding a new career path. I believe the key to success is really
            giving it your all in every opportunity you have through every
            single task, every single day, through hard work anything can be
            achieved.
            <br />
            <br />
            Since stepping down I've been working towards a new career path of
            becoming a web developer and for the past 5 months I've been a
            fast-track student at Coder Academy Melbourne where I'm honing my
            skills and knowelege so in a few months I will be ready to join the
            workforce as a junior developer.
            <br />
            <br />
            Through my expierences of life, work and study I've found my passion
            for technology, problem solving, collaboration and teamwork. I'm
            hoping through this new career path I can continue to persue these
            passions and work towards a career that can be more impactful.
            <br />
            <br />
            <br />
            <Button>
              <Link
                style={{ color: "white" }}
                to={resume}
                target="_blank"
                download
              >
                DOWNLOAD MY RESUME
              </Link>
            </Button>
          </p>
        </AboutContent>
      </AboutContainer>
    );
  }
}

export default AboutPage;
