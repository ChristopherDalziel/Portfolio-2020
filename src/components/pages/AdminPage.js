import React from "react";
import styled from "@emotion/styled";
import { connect } from "react-redux";
import { startLogin, startLogout } from "../../actions/auth";
import backgroundImage from "../../assets/Web_Background.png";
import { history } from "../../routers/AppRouter";

const LoginContainer = styled.main`
  background-image: url(${backgroundImage});
  background-size: cover;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Button = styled.a`
  font-family: "Open Sans", sans-serif;
  background-color: #ef636c;
  border: none;
  color: white;
  padding: 12px 15px;
  cursor: pointer;
  font-size: 15px;
  margin: 50px;
  &:hover {
    background: #a8b7ab;
    -webkit-transition: background 0.3s ease-in-out;
    -moz-transition: background 0.3s ease-in-out;
    transition: background-color 0.3s ease-in-out;
  }
`;

export const AdminPage = ({ isAuthenticated, startLogin, startLogout }) => {
  return (
    <LoginContainer>
      {isAuthenticated ? (
        <>
          <Button onClick={startLogout}>Logout</Button>
          <br />
          ||
          <br />
          <Button
            onClick={() => {
              history.push("/createproject");
            }}
          >
            Create new project
          </Button>
        </>
      ) : (
        <Button onClick={startLogin}>Login</Button>
      )}
    </LoginContainer>
  );
};

const mapStateToProps = (state) => ({
  isAuthenticated: !!state.auth.uid,
});

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startLogout: () => dispatch(startLogout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminPage);
