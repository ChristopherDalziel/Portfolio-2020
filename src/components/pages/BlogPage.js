import React, { Component } from "react";
import axios from "axios";
import styled from "@emotion/styled";
import backgroundImage from "../../assets/Web_Background.png";
import ArticlePreview from "../ArticlePreview";

const BlogPageContainer = styled.main`
  background-image: url(${backgroundImage});
  width: 100vw;
  height: 100vh;

  overflow: scroll;
`;

class BlogPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      posts: [],
    };
  }
  componentDidMount() {
    axios
      .get(
        "https://public-api.wordpress.com/rest/v1/sites/christopherdalziel88967465.wordpress.com/posts"
      )
      .then((res) => {
        console.log(res.data.posts);
        this.setState({ posts: res.data.posts });
      })
      .catch((error) => console.log(error));
  }

  render() {
    return (
      <BlogPageContainer className="blog">
        <div className="anchor" id="blog" />
        <h1 className="sectionTitle">Articles</h1>
        {this.state.posts.map((post) => (
          <ArticlePreview post={post} />
        ))}
      </BlogPageContainer>
    );
  }
}

export default BlogPage;
