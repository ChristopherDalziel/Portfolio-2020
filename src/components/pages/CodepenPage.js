import React from "react";

const CodepenPage = () => {
  return (
    <div>
      <p>Going to embed codepens here</p>
    </div>
  );
};

export default CodepenPage;
