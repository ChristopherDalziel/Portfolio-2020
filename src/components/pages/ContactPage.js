import React from "react";
import styled from "@emotion/styled";
import ContactForm from "../ContactForm";
import backgroundImage from "../../assets/Web_Background.png";

const ContactContainer = styled.div`
  background-image: url(${backgroundImage});
  background-size: cover;
  display: flex;
  flex-direction: column;
  background-color: transparent;
  height: 100vh;
`;

const ContactPage = () => {
  return (
    <ContactContainer>
      <ContactForm />
    </ContactContainer>
  );
};

export default ContactPage;
