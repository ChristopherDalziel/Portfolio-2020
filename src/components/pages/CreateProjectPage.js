import React from "react";
import styled from "@emotion/styled";
import { connect } from "react-redux";
import ProjectForm from "../ProjectForm";
import { startCreateProject } from "../../actions/projects";

const ProjectPageContainer = styled.main`
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const ProjectHeading = styled.h1`
  margin: 30px;
`;

export class CreateProjectPage extends React.Component {
  onSubmit = (project) => {
    this.props.startCreateProject(project);
    this.props.history.push("/portfolio");
  };

  render() {
    return (
      <ProjectPageContainer>
        <ProjectHeading>Create a new project.</ProjectHeading>
        <ProjectForm onSubmit={this.onSubmit} />
      </ProjectPageContainer>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateProject: (project) => dispatch(startCreateProject(project)),
});

export default connect(undefined, mapDispatchToProps)(CreateProjectPage);
