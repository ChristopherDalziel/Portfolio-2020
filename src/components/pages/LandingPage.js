import React from "react";
import styled from "@emotion/styled";
import backgroundImage from "../../assets/Web_Final_Col.jpg";
import mobileBackgroundImage from "../../assets/Web_Final_Col_Mob.jpg";
import { keyframes } from "@emotion/core";

const fadeInAnimation = keyframes`
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}`;

const LandingPageContainer = styled.main`
  display: flex;
  flex-direction: column;
  background-image: url(${backgroundImage});
  background-size: cover;
  height: 100vh;
  width: 100vw;
  @media (max-width: 815px) {
    background-image: url(${mobileBackgroundImage});
    background-size: auto 80vh;
    background-repeat: no-repeat;
`;

const HeroTextContainer = styled.div`
  animation: ${fadeInAnimation} ease 3s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  height: 80%;
  display: flex;
  flex-direction: column;
  background-color: transparent;
  color: #493a37;
  margin-left: 18%;
`;

const Heading = styled.h1`
  margin-top: 30%;
  font-size: 6em;
  @media (max-width: 815px) {
    margin-top: 60vh;
    font-size: 60px;
`;

const SubHeading = styled.h2`
  font-size: 3em;
  @media (max-width: 815px) {
    font-size: 25px;
`;

function LandingPage() {
  return (
    <LandingPageContainer>
      <HeroTextContainer>
        <Heading>HELLO.</Heading>
        <SubHeading>my name's Chris.</SubHeading>
      </HeroTextContainer>
    </LandingPageContainer>
  );
}

export default LandingPage;
