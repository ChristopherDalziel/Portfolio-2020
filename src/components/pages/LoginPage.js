import React from "react";
import styled from "@emotion/styled";
import { connect } from "react-redux";
import { startLogin, startLogout } from "../../actions/auth";

const LoginContainer = styled.main`
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const LoginButton = styled.a`
  font-family: "Open Sans", sans-serif;
  background-color: #ef636c;
  border: none;
  color: white;
  padding: 12px 15px;
  cursor: pointer;
  font-size: 15px;
  &:hover {
    background: #a8b7ab;
    -webkit-transition: background 0.3s ease-in-out;
    -moz-transition: background 0.3s ease-in-out;
    transition: background-color 0.3s ease-in-out;
  }
`;

export const LoginPage = ({ isAuthenticated, startLogin, startLogout }) => {
  return (
    <LoginContainer>
      {isAuthenticated ? (
        <LoginButton onClick={startLogout}>Logout</LoginButton>
      ) : (
        <LoginButton onClick={startLogin}>Login</LoginButton>
      )}

      {/* <p>
        : If you've found this page and you're not an admin please return to the
        rest of my portfolio by using the navigation above.
      </p> */}
    </LoginContainer>
  );
};

const mapStateToProps = (state) => ({
  isAuthenticated: !!state.auth.uid,
});

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startLogout: () => dispatch(startLogout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
