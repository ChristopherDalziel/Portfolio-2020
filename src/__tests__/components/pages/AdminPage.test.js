import React from "react";
import { mount, shallow, render } from "enzyme";
import toJSON from "enzyme-to-json";
import { AdminPage } from "../../../components/pages/AdminPage";

test("Should render Admin Page correctly", () => {
  const wrapper = shallow(<AdminPage />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

// Should be calling however is not working for unknown reasons
test("Should call startLogin on button click", () => {
  const startLoginSpy = jest.fn();
  const wrapper = mount(<AdminPage startLogin={startLoginSpy} />);
  wrapper.find("Button").simulate("click");
  expect(startLoginSpy).toHaveBeenCalled();
});
