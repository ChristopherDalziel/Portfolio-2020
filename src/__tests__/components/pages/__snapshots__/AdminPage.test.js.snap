// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Admin Page correctly 1`] = `
<Styled(main)>
  <Button>
    Login
  </Button>
</Styled(main)>
`;
